src/Compiler/Microsoft.CodeAnalysis.Razor/src/BindTagHelperDescriptorProvider.cs

```diff
@@ -262,20 +272,15 @@ private List<ElementBindData> GetElementBindData(Compilation compilation)
             }
         }
 
-        var results = new List<ElementBindData>();
+        using var results = new PooledArrayBuilder<ElementBindData>();
 
-        for (var i = 0; i < types.Count; i++)
+        foreach (var type in types)
         {
-            var type = types[i];
-            var attributes = type.GetAttributes();
-
             // Not handling duplicates here for now since we're the primary ones extending this.
             // If we see users adding to the set of 'bind' constructs we will want to add deduplication
             // and potentially diagnostics.
-            for (var j = 0; j < attributes.Length; j++)
+            foreach (var attribute in type.GetAttributes())
```

Comment:
Not sure if it happens, but this loop also could call ToDisplayString on the same type/type.ContainingNamespace multiple times.

Reply:
Thanks!